scraper.rb

```diff
@@ -33,6 +36,67 @@ def tidy
   end
 end
 
+class Mirror
+  include Capybara::DSL
+
+  attr_accessor :url
+
+  def save_page(url)
+    url = url.match(/(.*)_piref[\d_]+\.(next_page.*)/).captures.join('')
+    sha = Digest::SHA1.hexdigest url
+    path = File.join('.', CACHE_DIR, sha)
+
+    return if File.exist?(path)
+
+    File.open(path,"w") do |f|
+      f.write(page.html)
+    end
+    sleep(1)
```

Comment:
Isn't that taken care off with the 

```ruby
    # the website is a bit fragile to lets not hammer it with requests
    sleep(2)
```

below?

Sleeping in the loop seems fine (where anyone can choose their own approach), but forcing a sleep inside the `save_page` method doesn't seem right.